File: doc/src_OpenGL/Depot.cpp
// Depot.cpp: implementation of the Depot class.
//
//////////////////////////////////////////////////////////////////////
#include "Stdafx.h"
#include "OpenGL.h"
#include "Inkludy.h"

#ifdef _DEBUG
#undef THIS_FILE
static char THIS_FILE[]=__FILE__;
#define new DEBUG_NEW
#endif

//////////////////////////////////////////////////////////////////////
// Construction/Destruction
//////////////////////////////////////////////////////////////////////

Depot::Depot(Position* p, int slippery)
: WorldObject(WorldObjectVerifier::DEPOT,p,true,false,false,true)
{	
	if( efficiency < 0 )
		this->efficiency = 100;
	else 
		this->efficiency = efficiency;
	lighted = false;
}

Depot::~Depot()
{

}

bool Depot::conditionalMovement(WorldObject* const worldObject, 
			const int direction,const int MaxPower, int& usedPower)
{
	return true;
}

void Depot::evolve()
{
	lighted = !lighted;
}

bool Depot::getLighted()
{
	return lighted;
}

File: app/src/main/java/robot/object/depot.java
package robot.object;

import org.apache.commons.lang.mutable.MutableInt;
import robot.Position;
import robot.WorldObjectVerifier;

public class Depot extends WorldObject {

    private int efficiency;
    private boolean lighted;

    public Depot(Position p, int slippery) {
        super(WorldObjectVerifier.DEPOT.getIntValue(),p,true,false,false,true);
        if( efficiency < 0 )
            this.efficiency = 100;
        else
            this.efficiency = efficiency;
        lighted = false;
    }

    public boolean conditionalMovement(final WorldObject worldObject, final int direction, final int MaxPower, MutableInt usedPower)
    {
        return true;
    }

    public void evolve()
    {
        lighted = !lighted;
    }

    public boolean getLighted()
    {
        return lighted;
    }

}